plugin/audit_log/audit_log.c

```diff
@@ -219,6 +219,39 @@ void xml_escape(const char *in, size_t *inlen, char *out, size_t *outlen)
     { '\n', 5, "&#10;" },
     { '\t', 5, "&#9;" },
     { '"',  6, "&quot;" },
+    // escaping basic ASCII control characters
+    // skipping 0: it can't be represented in a valid XML document
+    { 1,  8, "\\u#x0001" },
+    { 2,  8, "\\u#x0002" },
+    { 3,  8, "\\u#x0003" },
+    { 4,  8, "\\u#x0004" },
+    { 5,  8, "\\u#x0005" },
+    { 8,  8, "\\u#x0006" },
+    { 7,  8, "\\u#x0007" },
+    { 8,  8, "\\u#x0008" },
+    { 9,  8, "\\u#x0009" },
```

Comment:
Is this OK that we have duplicates in this table?
`'\t'` == `9`
`'\n'` == `10`
`'\r'` == `13`

Reply:
I didn't want to delete the original entries (to reduce the chance of merge conflicts later), but I also thought that adding every entry to this list would be clearer than skipping those records.

Technically, it won't cause any problems, as the escape code will stop as soon as it finds an entry in this list. If I remove one of the variants, it could end up a bit faster, but now we are dealing with quite larger arrays than before (around 40 entries vs the 7-8 original). 

If this is a performance critical part of the server (I thought it isn't that critical, as it looks like a verbose logging feature), I could implement this differently, but that wouldn't be just the adding of a few lines to the existing escaping code, but changing parts of the actual escape loop.

Reply:
This is fully our code. Merge conflicts should not be a concern

Reply:
In this case, I updated the PR based no my original idea: I introduced a new array for the control characters, so we won't end up in a 40 element loop for each character.